Separate into Components

Kuheng opened the pull request.

Kuheng committed:
Separate into Components

kjinwoo12 commented on src/App.tsx:
```diff
@@ -54,114 +62,9 @@ function App() {
-        </div>
-      </div>
+      {sideMenu()}
+      {mainView(t, inputData, onChange_inputData, onClick_toggleLocales, colorPickerState, color, setColor, onClick_togglePickerState,
```
파라미터가 많아 이해하기 어렵습니다.

kjinwoo12 replied:
i18n 을 사용하기 위해 `const { t } = useTranslation()`를 한 번 선언하고 파라미터로 넘겨주어 재사용하는 것은 파라미터로넘길 때마다 객체가 복사되는 문제가 있습니다. GC에 의해 제거되지도 않아 수정이 필요해보입니다.

직접 @Kuheng 와 대화를 했고, 이미 @kuheng 은 문제를 확인해 필요한 곳마다 useTranslation으로 t를 선언해 사용하는 것으로 변경 중인 것을 확인했습니다.

kjinwoo12 commented on src/component/form/promptInputForm.tsx:
```diff
@@ -0,0 +1,15 @@
+function promptInputForm (t:any, inputData:any, type:string, onClick:Function, onChange:Function) {
+  return (
+    <>
+      <p className="w-auto h-auto text-">{t("main:"+type+"Prompt")}
```
t("main:"+type+"Prompt")와 
번역 관련 키값 같은데 따로 데이터 가공하는 것이 아니라면 문자열 연산을 하는 것보다는 직접 "imagePrompt", "negativePrompt" 등 값을 넣어주는게 더 좋아보입니다. promptInputForm 사용할 때 전달하는 데이터는 "image"와 "negative"인데 번역 데이터의 키값에는 두 값이 확인되지 않아 찾는데 혼란스러운 것 같습니다.

kjinwoo12 commented on src/component/form/seedInputForm.tsx:
```diff
@@ -0,0 +1,12 @@
+  return (
+    <div className="w-full h-18 text-start">
+      <p className="w-auto h-auto text-lg">{t("main:seedLabel")}</p>
+      <input type="range" value={seed} onChange={e=>{onChange("seed", e.target.value)}} min={0} max={4294967294}
```
시드 복사 붙여넣기 할 때 불편함이 있는 UI 같습니다. 텍스트 입력으로 바꿔도 좋을 듯 합니다.

kjinwoo12 commented on src/component/viewer/generateSettingViewer.tsx:
```diff
@@ -0,0 +1,21 @@
+    <div className="text-center w-300 h-auto mx-3">
+      {seedInputForm(inputData.seed, t, onChange)}
+      <div className="flex w-full h-auto">
+        {selectInputForm("format", t("main:formatLabel"), inputData.format, "auto", formatList, onChange)}
```
드롭다운만 따로 디자인을 위해 분리하고 라벨은 `selectInputForm 밖으로 빼는게 나중에 ui 수정사항이 생겨도 작업량이 줄어들 것 같습니다.

Kuheng committed:
Spearate into Component, Apply Recoil, Rename File, Function

Kuheng committed:
Add Prompt History by localHistory, Spearate Prompt History, Api Function Component

Kuheng committed:
Rename File

lowercase to uppercase

kjinwoo12 commented:
1:1 통화로 코드리뷰 했습니다. 전체적인 구조의 변경은 없을 예정이나 자잘한 네임 컨벤션과 코드 스타일을 변경하는 것으로 이야기를 했습니다.

Kuheng committed:
Apply Feedback, Add Button Image

applying feedback from 2024-07-03 code review
- combine component(ApiKeyInput, QuantityInput)
- inlilne function to local function
- rename function
- separate into states
